Mathlib/CategoryTheory/Functor/FunctorHom.lean

```diff
@@ -0,0 +1,151 @@
+/-
+Copyright (c) 2024 Jack McKoen. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jack McKoen, Joël Riou
+-/
+import Mathlib.CategoryTheory.Monoidal.FunctorCategory
+import Mathlib.CategoryTheory.Monoidal.Types.Basic
+
+/-!
+# Internal hom in functor categories
+
+Given functors `F G : C ⥤ D`, define a functor `functorHom F G` from `C` to `Type max v' v u`,
+which is a proxy for the "internal hom" functor Hom(F ⊗ coyoneda(-), G). This is useful for
+showing that the functor category `C ⥤ D` is enriched over `C ⥤ Type max v' v u`, and
+that `C ⥤ Type max w v u` is monoidal closed.
+
+See `Mathlib.CategoryTheory.Closed.FunctorToTypes` and `Mathlib.CategoryTheory.Enriched.Functor`.
+
+-/
+
+
+universe w v' v u u'
+
+open CategoryTheory MonoidalCategory
+
+namespace CategoryTheory.Functor
+
+variable {C : Type u} [Category.{v} C] {D : Type u'} [Category.{v'} D]
+
+variable (F G : C ⥤ D)
+
+/-- Given functors `F G : C ⥤ D`, `HomObj F G A` is a proxy for the type
+of "morphisms" `F ⊗ A ⟶ G`, where `A : C ⥤ Type w` (`w` an arbitrary universe). -/
+@[ext]
+structure HomObj (A : C ⥤ Type w) where
+  /-- The morphism `F.obj c ⟶ G.obj c` associated with `a : A.obj c`. -/
+  app (c : C) (a : A.obj c) : F.obj c ⟶ G.obj c
+  naturality {c d : C} (f : c ⟶ d) (a : A.obj c) :
+    F.map f ≫ app d (A.map f a) = app c a ≫ G.map f := by aesop_cat
+
+/-- When `F`, `G`, and `A` are all functors `C ⥤ Type w`, then `HomObj F G A` is in
+bijection with `F ⊗ A ⟶ G`. -/
+def HomObjEquiv (F G A : C ⥤ Type w) : (F ⊗ A ⟶ G) ≃ (HomObj F G A) where
```

Comment:
The naming convention says that this should be `homObjEquiv`. Also, I think it would be more common to put `HomObj F G A` on the left of the equivalence.